pyfr/denserkschemes/__init__.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+
+from pkg_resources import resource_listdir, resource_string
+import re
+
+import numpy as np
+
+
+class BaseTabulatedDensePseudoStepper(object):
+    def __init__(self, rule):
+        a = []
+
+        rule = re.sub(r'(?<=\))\s*,?\s*(?!$)', r'\n', rule)
+        rule = re.sub(r'\(|\)|,', '', rule).strip()
+        rule = rule[1:-1] if rule.startswith('[') else rule
+
+        for l in rule.splitlines()[:-1]:
+            if not l:
+                continue
+
+            # Store the Butcher A matrix
+            a.append([float(f) for f in l.split()])
+
+        # The last line is the b vector
+        b = [float(f) for f in rule.splitlines()[-1].split()]
+
+        # Cast
+        self.a = np.array(a, dtype=np.float)
+        self.b = np.array(b, dtype=np.float)
+
+
+class BaseDensePseudoStepper(BaseTabulatedDensePseudoStepper):
+    @classmethod
+    def _iter_scheme(cls):
+        rpaths = getattr(cls, '_rpaths', None)
+        if rpaths is None:
+            cls._rpaths = rpaths = resource_listdir(__name__, cls.pstype)
+
+        for path in rpaths:
+            m = re.match(r'([a-zA-Z0-9\-~+]+)-s(\d+)'
+                         r'(?:-p(\d+))?\.txt$', path)
```

Comment:
p might be confusing as it could also refer to the order of accuracy of the scheme.  Should probably settle on a suffix for denoting the optimal spatial polynomial order.